crates/swc_css_ast/src/selector.rs

```diff
@@ -130,12 +131,48 @@ pub struct AttrSelector {
     pub modifier: Option<char>,
 }
 
+#[ast_node]
+#[derive(Is)]
+pub enum PseudoSelectorChildren {
+    #[tag("Nth")]
+    Nth(Nth),
+
+    #[tag("Tokens")]
+    Tokens(Tokens),
+}
+
+#[ast_node("Nth")]
+pub struct Nth {
+    pub span: Span,
+    pub nth: NthValue,
+    pub selector_list: Option<SelectorList>,
+}
+
+#[ast_node("AnPlusB")]
+pub struct AnPlusB {
+    pub span: Span,
+    pub a: Option<i32>,
+    pub a_raw: Option<JsWord>,
+    pub b: Option<i32>,
+    pub b_raw: Option<JsWord>,
+}
+
+#[ast_node]
+#[derive(Is)]
+pub enum NthValue {
+    #[tag("AnPlusB")]
+    AnPlusB(AnPlusB),
+
+    #[tag("Ident")]
+    Ident(Ident),
+}
+
 #[ast_node("PseudoSelector")]
 pub struct PseudoSelector {
     pub span: Span,
     pub is_element: bool,
     pub name: Ident,
-    pub args: Tokens,
+    pub children: Option<PseudoSelectorChildren>,
```

Comment:
Is empty `Tokens` not enough?

Reply:
@kdy1 we will have more different types here in future, for example `:is` will contains selector list, other normal values, also some of them do not contains `args`, for example `:focus {}`, so `children` (previously `args`) is optional

Reply:
We do the same with any optional fields for our AST Nodes

Reply:
Thanks for description!